core/src/tangram.h

```diff
@@ -11,12 +11,27 @@ namespace Tangram {
 
 class DataSource;
 
+enum LabelType {
+    icon,
+    text,
+};
+
 struct TouchItem {
     std::shared_ptr<Properties> properties;
     float position[2];
     float distance;
 };
 
+using FeatureSelectionCallback = std::function<void(const std::vector<TouchItem>&)>;
+
+struct TouchLabel {
+    LabelType type;
+    LngLat coordinate;
+    TouchItem touchItem;
+};
+
+using LabelSelectionCallback = std::function<void(const std::vector<TouchLabel>&)>;
```

Comment:
Is it possible that more than one label could be selected for a query at a single pixel? If not then we could return just one label pointer or nullptr.